packages/webapp/src/components/RepeatCropPlan/utils.js

```diff
@@ -0,0 +1,187 @@
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { RRule } from 'rrule';
+import { getRruleLanguage } from '../../util/rruleTranslation';
+import { getLanguageFromLocalStorage } from '../../util/getLanguageFromLocalStorage';
+import { parseISOStringToLocalDate } from '../Form/Input/utils';
+
+const RRULEDAYS = {
+  Sunday: 'SU',
+  Monday: 'MO',
+  Tuesday: 'TU',
+  Wednesday: 'WE',
+  Thursday: 'TH',
+  Friday: 'FR',
+  Saturday: 'SA',
+};
+
+const translationCache = {};
+
+// Returns rrule language definition if already imported, otherwise loads and saves it to memory
+const getTranslations = async (lang) => {
+  if (translationCache[lang]) {
+    return translationCache[lang];
+  }
+  const translation = await getRruleLanguage(lang);
+  translationCache[lang] = translation;
+  return translation;
+};
+
+export const getWeekday = (planStartDate) => {
+  const date = new Date(parseISOStringToLocalDate(planStartDate));
+
+  return date.toLocaleString('en', { weekday: 'long' });
+};
+
+export const getDate = (planStartDate) => {
+  const date = new Date(parseISOStringToLocalDate(planStartDate));
+
+  return date.getDate();
+};
+
+const calculateWeekdayOrdinal = (date) => {
+  const dayOfMonth = date.getDate();
+
+  let ordinal = Math.ceil(dayOfMonth / 7);
+
+  // Check if the day is within the last 7 days of month
+  const lastDayOfMonth = new Date(date.getFullYear(), date.getMonth() + 1, 0);
+  if (dayOfMonth > lastDayOfMonth.getDate() - 7) {
+    // If within last week, return -1 as the ordinal
+    ordinal = -1;
+  }
+
+  return ordinal;
+};
+
+export const calculateMonthlyOptions = async (planStartDate, repeatFrequency) => {
+  const currentLang = getLanguageFromLocalStorage();
+
+  const translations = await getTranslations(currentLang);
+
+  const dt = parseISOStringToLocalDate(planStartDate);
+  const weekday = RRULEDAYS[dt.toLocaleString('en', { weekday: 'long' })];
+  const date = dt.getDate();
+  const ordinal = calculateWeekdayOrdinal(dt);
+
+  const dateRule = new RRule({
+    freq: RRule.MONTHLY,
+    dtstart: dt,
+    bymonthday: [date],
+    interval: Number(repeatFrequency),
+  });
+
+  const dateString = dateRule.toText(translations.getText, translations.language);
+
+  const dayWeekRule = new RRule({
+    freq: RRule.MONTHLY,
+    dtstart: dt,
+    bymonthday: [],
+    byweekday: [RRule[weekday].nth(ordinal)],
+    interval: Number(repeatFrequency),
+  });
+
+  const dayWeekString = dayWeekRule.toText(translations.getText, translations.language);
+
+  return [
+    {
+      value: date,
+      label: dateString,
+    },
+    {
+      value: { ordinal, weekday },
+      label: dayWeekString,
+    },
+  ];
+};
+
+export const countOccurrences = ({
```

Comment:
Do you think you can return the array of occurrences here and do `setValue()` in the component so that I can use it in my component? (I plan to forward it to the backend)
I might have a few more requests like this, but I'm not sure if I should just make some updates after this PR is merged...

Reply:
Definitely! But I only generate an array of occurrences for the finish on date situation right now (not for the "finish after _x_ repetitions") and I tie it specifically to the validation of the date picker, so there would have to be some changes to that timing 🤔 

I am thinking then running the count onSubmit would be best then? Since beyond validation there is no need for it to run onChange. So the count function could be generalized and called in two situations -- the useEffect and onSubmit.

And YES you are so very welcome to adjust the heck out of it for your own uses and just change whatever you like for the hookup to your component!! That might be easier.

Reply:
Thank you Joyce! I wouldn't have felt good about modifying your code, but since you are good with that, you can leave the function as is! I will make adjustments later and ask you for a review.

I realized that my request was not as easy as I thought, so I will take what you mentioned into account and make a decision when I work on my ticket 🙏 🙏 🙏